nixos/modules/misc/mandoc.nix

```diff
@@ -36,28 +36,39 @@ in {
           configuration options used for the package.
         '';
       };
+
+      config = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        description = lib.mdDoc ''
+          Configuration to write to {manpage}`man.conf(5)`.
+        '';
+      };
     };
   };
 
   config = lib.mkIf cfg.enable {
     environment = {
       systemPackages = [ cfg.package ];
 
-      # tell mandoc about man pages
-      etc."man.conf".text = lib.concatMapStrings (path: ''
-        manpath /run/current-system/sw/${path}
-      '') cfg.manPath;
+      etc."man.conf".text = cfg.config;
 
       # create mandoc.db for whatis(1), apropos(1) and man(1) -k
       # TODO(@sternenseemman): fix symlinked directories not getting indexed,
       # see: https://inbox.vuxu.org/mandoc-tech/20210906171231.GF83680@athene.usta.de/T/#e85f773c1781e3fef85562b2794f9cad7b2909a3c
       extraSetup = lib.mkIf config.documentation.man.generateCaches ''
-        ${makewhatis} -T utf8 ${
+        for man_path in ${
           lib.concatMapStringsSep " " (path:
             "$out/" + lib.escapeShellArg path
           ) cfg.manPath
```

Comment:
```suggestion
          lib.escapeShellArgs (map (path: "$out/${path}") cfg.manPath)
```

Reply:
Then the shell substitution of `$out` would cease to work…

Reply:
> Then the shell substitution of `$out` would cease to work…

Yes.